src/ClusterBootstrap/az_tools.py

```diff
@@ -523,7 +523,8 @@ def create_vm_param(i, role, vm_size, no_az=False, arm_vm_password=None):
     elif role == "infra":
         vmname = "%s-infra%02d" % (config["azure_cluster"]
                                    ["cluster_name"], i + 1)
-    elif role == "dev":
+    else:
+        role = "dev"
```

Comment:
why is dev role set to default?

Reply:
From my perspective, assertion should be a better plan. Anbang pointed out that original code used dev as default, so here using dev as default is consistent with historical code. If you think it fine to use assertion(so we're guaranteed that users select valid machine roles.